### PR TITLE
Add k8sd daemon and introduce clustering command

### PR DESCRIPTION
**Pull request by bschimke95**
### Summary

Integrates `k8sd` as the MicroCluster management daemon on a node and expands the `k8s` CLI with the following commands:

    * `bootstrap-cluster`: Facilitates the creation of a new cluster
    * `list-cluster`: Provides information about cluster members

### Test
Manually tested for now - unittests/integration tests will be done in an other card.

`k8sd:`
```
➜ go build -o k8sd cmd/main.go && ./k8sd --state-dir /home/benjamin/k8sd-db/ 
INFO   [2023-11-24T11:04:26+01:00]  - binding control socket                     socket=/home/benjamin/k8sd-db/control.socket
INFO   [2023-11-24T11:04:26+01:00]  - binding https socket                       network="192.168.178.69:6443"
INFO   [2023-11-24T11:04:26+01:00] Preparing statements for Go project "k8sd"
```

`k8s bootstrap-cluster`:
```
➜ go build -o k8s cmd/main.go && ./k8s bootstrap-cluster --state-dir /home/benjamin/k8sd-db1/
INFO[0000] Cluster created.
```

`k8s list-cluster`:
```
➜ go build -o k8s cmd/main.go && ./k8s list-cluster --state-dir /home/benjamin/k8sd-db1/
+--------+---------------------+-------+------------------------------------------------------------------+--------+
|  NAME  |       ADDRESS       | ROLE  |                           FINGERPRINT                            | STATUS |
+--------+---------------------+-------+------------------------------------------------------------------+--------+
| carbon | 192.168.178.69:6443 | voter | 95d252423a64f71e1448c42a3f6cbb29773e17bb5c70944ae0ccb6107187bef8 | ONLINE |
+--------+---------------------+-------+------------------------------------------------------------------+--------+
```

### CODE REVIEW

File: src/k8s/pkg/k8s/cluster/cluster.go [hunk 1]
```diff
@@ -0,0 +1,170 @@
...
+
+	members := make([]ClusterMember, len(clusterMembers))
+	for i, clusterMember := range clusterMembers {
+		fingerprint, err := shared.CertFingerprintStr(clusterMember.Certificate.String())
+		if err != nil {
+			continue
+		}
+
+		members[i] = ClusterMember{
+			clusterMember.Name,
```
Comment (neoaggelos) [inline src/k8s/pkg/k8s/cluster/cluster.go:113]: Use named fields instead, also for other fields here.
```suggestion
			Name: clusterMember.Name,
```

File: src/k8s/pkg/k8s/cluster/cluster.go [hunk 2]
```diff
@@ -0,0 +1,170 @@
...
+	// Directory containing cluster information (for local clients)
+	StorageDir string
+	// Cluster address (for remote clients)
+	Address string
+	// Port on which the REST-API is exposed
+	Port string
+	// Enable info level logging
+	Verbose bool
+	// Enable trace level logging
+	Debug bool
```
Comment (neoaggelos) [inline src/k8s/pkg/k8s/cluster/cluster.go:145]: godoc comments should begin with the name of the field, e.g.
```
// StorageDir is the cluster state directory.
```

Also, nit; please end comments with a `.`

File: src/k8s/pkg/k8s/cluster/cluster.go [hunk 3]
```diff
@@ -0,0 +1,170 @@
...
+	}, nil
+}
+
+// Client is a wrapper around the MicroCluster client
+type Client struct {
+	microClient *client.Client
+}
+
+// GetMembers returns information about all members of the cluster.
+func (c *Client) GetMembers(ctx context.Context) ([]ClusterMember, error) {
```
Comment (neoaggelos) [inline src/k8s/pkg/k8s/cluster/cluster.go:99]: I much prefer `[]*ClusterMember` here

File: src/k8s/pkg/k8s/cluster/cluster.go [hunk 4]
```diff
@@ -0,0 +1,170 @@
...
+	"time"
+
+	"github.com/canonical/lxd/lxd/util"
+	"github.com/canonical/lxd/shared"
+	"github.com/canonical/microcluster/client"
+	"github.com/canonical/microcluster/microcluster"
+)
+
+// Bootstrap sets up new cluster and returns the information about the daemon.
+func Bootstrap(ctx context.Context, opts ClusterOpts) (ClusterMember, error) {
```
Comment (neoaggelos) [inline src/k8s/pkg/k8s/cluster/cluster.go:19]: I prefer returning a `*ClusterMember` instead, to minimize `copy`ing things around.

File: src/k8s/pkg/k8s/cluster/cluster.go (hunk 1 above)
Comment (bschimke95) [reply src/k8s/pkg/k8s/cluster/cluster.go]: Done

File: src/k8s/pkg/k8s/cluster/cluster.go (hunk 2 above)
Comment (bschimke95) [reply src/k8s/pkg/k8s/cluster/cluster.go]: Done

**Review by neoaggelos — APPROVED** [review]

**Pull request merged**